### PR TITLE
task/FP-1906 View Registrations hotfix - format phone number in modal

### PR DESCRIPTION
**Pull request by edmondsgarrett**
## Overview
Added function to format phone number for easier reading
…

## Related

…

## Changes
- Added function within contacts util function to format number
…

## Testing

1. In `src/apps/admin_regis_table/views.py`, replace lines 21-23 with the following snippet:
     <details><summary>Snippet</summary>
     
          import datetime
          registrations_content = [
            (
                1,                                  #registration_id
                datetime.date(2022, 8, 3),          #posted_date
                12023,                              #applicable_period_start
                122023,                             #applicable_period_end
                True,                               #file_me
                True,                               #file_pv
                True,                               #file_mc
                True,                               #file_pc
                False,                              #file_dc
                True,                               #submitting_for_self
                'SFTP',                             #submission_method
                'active',                           #registration_status
                'insurance carrier',                #org_type
                'Golden Rule Insurance Company',    #business_name
                '7440 Woodland Drive',              #mail_address
                'Indianpolis',                      #city
                'IN',                               #state
                '46278     '                        #zip
            ),
            (
                2,                                  #registration_id
                datetime.date(2022, 8, 3),          #posted_date
                12023,                              #applicable_period_start
                122023,                             #applicable_period_end
                True,                               #file_me
                True,                               #file_pv
                True,                               #file_mc
                True,                               #file_pc
                False,                              #file_dc
                True,                               #submitting_for_self
                'SFTP',                             #submission_method
                'active',                           #registration_status
                'insurance carrier',                #org_type
                'Golden Rule Insurance Company',    #business_name
                '7440 Woodland Drive',              #mail_address
                'Indianpolis',                      #city
                'IN',                               #state
                '46278     '                        #zip
            )
        ]
        registrations_entities = [
            (
                5, 1, 5, 46, 11111, 0, 5, 'Garretts Test Business 2', '11-0001111'
            ),
            (
                5, 2, 50000, 47, 1010, 1101, 1, 'A Second Test Entity', '00-0000000'
            )
        ]
        registrations_contacts = [
            (
                52,
                1,
                False,
                'Test Role',
                'Garrett Test Tester',
                '2222222222',
                'notarealemail@emailplace.email'
            ),
            (
                53,
                1,
                False,
                'A 2nd Test Role',
                'Test Garrett 2 Test',
                '145555555555',
                'testemail@testemail.emailtest'
            ),
            (
                54,
                2,
                True,
                'A 3rd and final test role',
                'Test 3rd Role Name Garrett',
                '0000000000',
                None
            )
          ]
        
    </details>
2. Open http://localhost:8000/administration/list-registration-requests/
3. Click on the 'View' button in the last column of either row
4. Confirm 10-digit phone numbers only have hyphens, and that phone numbers with more than 10 digits have a `+` in front and a space separating the country code and the 10-digit phone number

## UI
Before: 
image
After: 
image

…

### CODE REVIEW

**Comment by wesleyboar** [comment]
This fix also has opening the correct data in modal fix right? (I may have misinterpreted code to come up with that belief.)

**Comment by edmondsgarrett** [comment]
> This [PR] also has opening the correct data in modal fix right? (I may have misinterpreted code to come up with that belief.)

Thanks for that note! Had neglected to merge main, diff is now updated.

**Review by wesleyboar — APPROVED** [review]

File: apcd-cms/src/apps/admin_regis_table/views.py [hunk 1]
```diff
@@ -56,11 +56,22 @@ def _set_entities(reg_ent):
                 'fein': reg_ent[8]
             }
         def _set_contacts(reg_cont):
+
+            def format_phone_number(num):
+                formatted_num_list = list(reversed(num))
+                for n in range(len(formatted_num_list)):
+                    if n+1 == 4 or n+1 == 8:
```
Comment (fnets) [inline apcd-cms/src/apps/admin_regis_table/views.py:63]: Would this be more readable? If not, feel free to ignore me
```suggestion
                    if n == 3 or n == 7:
```

File: apcd-cms/src/apps/admin_regis_table/views.py [hunk 2]
```diff
@@ -56,11 +56,22 @@ def _set_entities(reg_ent):
...
+                        formatted_num_list.insert(n+1,'-')
+                    if n+1 == 12 and len(num) > 10:
```
Comment (fnets) [inline apcd-cms/src/apps/admin_regis_table/views.py:65]: Same readability suggestion, same invitation to ignore me
```suggestion
                    if n == 11 and len(num) > 10:
```

**Review by fnets — APPROVED** [review]
LGTM and works locally! Made some code suggestions, and they seem to work, too. Great job and quick turnaround!

File: apcd-cms/src/apps/admin_regis_table/views.py (hunk 2 above)
Comment (wesleyboar) [reply apcd-cms/src/apps/admin_regis_table/views.py:65]: [Same follow-up suggestion as the other.](https://github.com/TACC/Core-CMS-Custom/pull/52#discussion_r1028540039)

**Pull request merged**